400-claims.md

```diff
@@ -86,24 +98,12 @@ The fields above are defined as follows:
 
 - `action` (REQUIRED): The name of the action that last caused this claim to be updated. This may be any of the built-in actions (`install`, `upgrade`, `uninstall`) as well as any custom actions as defined in the bundle descriptor. The special name `unknown` MAY be used in the case where the CNAB Runtime cannot determine the action name of a claim.
 - `bundle` (REQUIRED): The bundle, as defined in [the Bundle Definition](101-bundle-json.md).
-- `created` (OPTIONAL): A timestamp indicating when this release claim was first created. This MUST not be changed after initial creation.
+- `created` (REQUIRED): A timestamp indicating when this claim was first created. This MUST not be changed after initial creation and does not change with new revisions.
+- `modified` (REQUIRED): A timestamp indicating the last time this claim was modified. Executing the installation action MUST set this to a timestamp that matches the claim's `created`. Executing any modifying action (`upgrade`, `uninstall`, or a custom action with the property `"modifies": true`) MUST set this field to the time of the operation.
```

Comment:
This seems to be the last mutable field on claims, and it seems like the `started` and `stopped` fields on result objects seem like they encode the same information. Could we remove `modified` and specify that claims are immutable instead?

Reply:
I think that modified is a field that could be used in different ways depending on how the tool stores claim documents. (See [my comment above](https://github.com/cnabio/cnab-spec/pull/349#discussion_r390486137) for more context on the two ways to store documents. I kept a modified field because of questions I received about removing modified entirely from the claim document, there wasn't clear consensus about making it immutable.

I believe this leaves enough room each implementer to decide if it wants to treat the claim document as immutable or not (overwriting the claim document on each revision or persisting each revision as a separate document)? Otherwise, let's talk about this at the next meeting and have people talk it out. 😀

Reply:
As I read the spec in its current version, this field can be treated as immutable.

- Install sets both to the same
- Upgrade clones the install time, and sets modified to the current time
- A non-destructive op clones both.

Reply:
Sorry for the late reply, just noticed this thread had more activity.

I'm a tiny bit confused now. Are we saying that `claim.created` represents the time of creation for the installation itself? I was thinking this was the time the claim was created. My working model is that the time of creation for the installation would be derived from the `claim.created` field of the claim with the earliest known revision.

Claim results _also_ have `created` and `modified` fields. This makes me question having `modified` on claims as well, because there is no standard way to answer questions like "when was this action completed". A runtime may choose do one of

- treat claims and results as immutable, returning the last known `result.created` timestamp associated with the claim
- treat claims as immutable but results as mutable, returning the `result.modified` timestamp
- treat claims as mutable, returning the `claim.modified`

I think more clarity on this from the spec would help improve the chances that implementers are able to build on the same core libraries in the future.